### PR TITLE
Update openjdk to non-headless on non-slim images

### PR DESCRIPTION
**Pull request by yosifkit**
Add test for openjdk non-headless: uimanager-font

contains: https://github.com/docker-library/openjdk/pull/143

### CODE REVIEW

**Comment by yosifkit** [comment]
<details>
<summary>Diff:</summary>

```diff
diff --git a/openjdk_6-jre/Dockerfile b/openjdk_6-jre/Dockerfile
index 70c70d2..11d4c03 100644
--- a/openjdk_6-jre/Dockerfile
+++ b/openjdk_6-jre/Dockerfile
@@ -51,7 +51,7 @@ RUN set -ex; \
 	\
 	apt-get update; \
 	apt-get install -y \
-		openjdk-6-jre-headless="$JAVA_DEBIAN_VERSION" \
+		openjdk-6-jre="$JAVA_DEBIAN_VERSION" \
 	; \
 	rm -rf /var/lib/apt/lists/*; \
 	\
diff --git a/openjdk_7-jre/Dockerfile b/openjdk_7-jre/Dockerfile
index 9be1fb6..5304ad1 100644
--- a/openjdk_7-jre/Dockerfile
+++ b/openjdk_7-jre/Dockerfile
@@ -51,7 +51,7 @@ RUN set -ex; \
 	\
 	apt-get update; \
 	apt-get install -y \
-		openjdk-7-jre-headless="$JAVA_DEBIAN_VERSION" \
+		openjdk-7-jre="$JAVA_DEBIAN_VERSION" \
 	; \
 	rm -rf /var/lib/apt/lists/*; \
 	\
diff --git a/openjdk_9-jre/Dockerfile b/openjdk_9-jre/Dockerfile
index 3467092..1013e49 100644
--- a/openjdk_9-jre/Dockerfile
+++ b/openjdk_9-jre/Dockerfile
@@ -51,7 +51,7 @@ RUN set -ex; \
 	\
 	apt-get update; \
 	apt-get install -y \
-		openjdk-9-jre-headless="$JAVA_DEBIAN_VERSION" \
+		openjdk-9-jre="$JAVA_DEBIAN_VERSION" \
 	; \
 	rm -rf /var/lib/apt/lists/*; \
 	\
diff --git a/openjdk_9/Dockerfile b/openjdk_9/Dockerfile
index 050ea69..20ab93b 100644
--- a/openjdk_9/Dockerfile
+++ b/openjdk_9/Dockerfile
@@ -51,7 +51,7 @@ RUN set -ex; \
 	\
 	apt-get update; \
 	apt-get install -y \
-		openjdk-9-jdk-headless="$JAVA_DEBIAN_VERSION" \
+		openjdk-9-jdk="$JAVA_DEBIAN_VERSION" \
 	; \
 	rm -rf /var/lib/apt/lists/*; \
 	\
diff --git a/openjdk_jre/Dockerfile b/openjdk_jre/Dockerfile
index 01c525a..33ff75c 100644
--- a/openjdk_jre/Dockerfile
+++ b/openjdk_jre/Dockerfile
@@ -55,7 +55,7 @@ RUN set -ex; \
 	\
 	apt-get update; \
 	apt-get install -y \
-		openjdk-8-jre-headless="$JAVA_DEBIAN_VERSION" \
+		openjdk-8-jre="$JAVA_DEBIAN_VERSION" \
 		ca-certificates-java="$CA_CERTIFICATES_JAVA_VERSION" \
 	; \
 	rm -rf /var/lib/apt/lists/*; \
```

</details>

**Comment by yosifkit** [comment]
Build test of #3424; 30c01463 (`openjdk`):

```console

$ bashbrew build openjdk:6b38-jdk
Using bashbrew/cache:2736b530dc887cb71398e072a20a8da28302d3273c7441f0e90cbf149f6b8610 (openjdk:6b38-jdk)
Tagging openjdk:6b38-jdk
Tagging openjdk:6b38
Tagging openjdk:6-jdk
Tagging openjdk:6

$ test/run.sh openjdk:6b38-jdk
testing openjdk:6b38-jdk
	'utc' [1/6]...passed
	'cve-2014--shellshock' [2/6]...passed
	'no-hard-coded-passwords' [3/6]...passed
	'override-cmd' [4/6]...passed
	'java-hello-world' [5/6]...passed
	'java-uimanager-font' [6/6]...passed


$ bashbrew build openjdk:6b38-jdk-slim
Using bashbrew/cache:3c0cb44050c0daf0bcb2def1a43cf9e4296d334f3556cabef26169648c445a32 (openjdk:6b38-jdk-slim)
Tagging openjdk:6b38-jdk-slim
Tagging openjdk:6b38-slim
Tagging openjdk:6-jdk-slim
Tagging openjdk:6-slim

$ test/run.sh openjdk:6b38-jdk-slim
testing openjdk:6b38-jdk-slim
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'java-hello-world' [5/5]...passed


$ bashbrew build openjdk:6b38-jre
Building bashbrew/cache:de4d991273bc6b957594221e49300579ef96adaec04a8ef45c24df3f8a069c0d (openjdk:6b38-jre)
Tagging openjdk:6b38-jre
Tagging openjdk:6-jre

$ test/run.sh openjdk:6b38-jre
testing openjdk:6b38-jre
	'utc' [1/6]...passed
	'cve-2014--shellshock' [2/6]...passed
	'no-hard-coded-passwords' [3/6]...passed
	'override-cmd' [4/6]...passed
	'java-hello-world' [5/6]...passed
	'java-uimanager-font' [6/6]...passed


$ bashbrew build openjdk:6b38-jre-slim
Using bashbrew/cache:25594aa438409e5a4a4f4fb7c6343bb1cffe4833caa071667ea1ef59b7176386 (openjdk:6b38-jre-slim)
Tagging openjdk:6b38-jre-slim
Tagging openjdk:6-jre-slim

$ test/run.sh openjdk:6b38-jre-slim
testing openjdk:6b38-jre-slim
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'java-hello-world' [5/5]...passed


$ bashbrew build openjdk:7u151-jdk
Using bashbrew/cache:2b71666dcf967cc248f43e23d8ed519f970e3bfca0b431750f9b0cd02724b459 (openjdk:7u151-jdk)
Tagging openjdk:7u151-jdk
Tagging openjdk:7u151
Tagging openjdk:7-jdk
Tagging openjdk:7

$ test/run.sh openjdk:7u151-jdk
testing openjdk:7u151-jdk
	'utc' [1/6]...passed
	'cve-2014--shellshock' [2/6]...passed
	'no-hard-coded-passwords' [3/6]...passed
	'override-cmd' [4/6]...passed
	'java-hello-world' [5/6]...passed
	'java-uimanager-font' [6/6]...passed


$ bashbrew build openjdk:7u151-jdk-slim
Using bashbrew/cache:2447d2c89cb815d8064e7288f0298225b054eeb80e7067617b446d2a6f2dafca (openjdk:7u151-jdk-slim)
Tagging openjdk:7u151-jdk-slim
Tagging openjdk:7u151-slim
Tagging openjdk:7-jdk-slim
Tagging openjdk:7-slim

$ test/run.sh openjdk:7u151-jdk-slim
testing openjdk:7u151-jdk-slim
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'java-hello-world' [5/5]...passed


$ bashbrew build openjdk:7u131-jdk-alpine
Using bashbrew/cache:955b9c8c4dfdea0f8c50daa9a87a1d23456cec02ed97d92c06f4cf5d9529f612 (openjdk:7u131-jdk-alpine)
Tagging openjdk:7u131-jdk-alpine
Tagging openjdk:7u131-alpine
Tagging openjdk:7-jdk-alpine
Tagging openjdk:7-alpine

$ test/run.sh openjdk:7u131-jdk-alpine
testing openjdk:7u131-jdk-alpine
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'java-hello-world' [5/5]...passed


$ bashbrew build openjdk:7u151-jre
Building bashbrew/cache:4013980ec856ee6c033ba353c7f47a4d0f76c84139cf6a7d0674a0a687fe6022 (openjdk:7u151-jre)
Tagging openjdk:7u151-jre
Tagging openjdk:7-jre

$ test/run.sh openjdk:7u151-jre
testing openjdk:7u151-jre
	'utc' [1/6]...passed
	'cve-2014--shellshock' [2/6]...passed
	'no-hard-coded-passwords' [3/6]...passed
	'override-cmd' [4/6]...passed
	'java-hello-world' [5/6]...passed
	'java-uimanager-font' [6/6]...passed


$ bashbrew build openjdk:7u151-jre-slim
Using bashbrew/cache:bf21b258f0f38def6ea74d65c38c97bcf999d7e195c3d79fb5d22bf8892f2528 (openjdk:7u151-jre-slim)
Tagging openjdk:7u151-jre-slim
Tagging openjdk:7-jre-slim

$ test/run.sh openjdk:7u151-jre-slim
testing openjdk:7u151-jre-slim
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'java-hello-world' [5/5]...passed


$ bashbrew build openjdk:7u131-jre-alpine
Using bashbrew/cache:8b835c5a87839dff9d39b8c0215bb8a94df78acb8dc7ed3cfc375ba8c0c0eb5c (openjdk:7u131-jre-alpine)
Tagging openjdk:7u131-jre-alpine
Tagging openjdk:7-jre-alpine

$ test/run.sh openjdk:7u131-jre-alpine
testing openjdk:7u131-jre-alpine
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'java-hello-world' [5/5]...passed


$ bashbrew build openjdk:8u141-jdk
Using bashbrew/cache:0e495352767c3561271c022783ad01e98c72e2e7cda43897049a1fe94bba2dc8 (openjdk:8u141-jdk)
Tagging openjdk:8u141-jdk
Tagging openjdk:8u141
Tagging openjdk:8-jdk
Tagging openjdk:8
Tagging openjdk:jdk
Tagging openjdk:latest

$ test/run.sh openjdk:8u141-jdk
testing openjdk:8u141-jdk
	'utc' [1/6]...passed
	'cve-2014--shellshock' [2/6]...passed
	'no-hard-coded-passwords' [3/6]...passed
	'override-cmd' [4/6]...passed
	'java-hello-world' [5/6]...passed
	'java-uimanager-font' [6/6]...passed


$ bashbrew build openjdk:8u141-jdk-slim
Using bashbrew/cache:1e13f0b606d9201d382cf16924baccaf4399d9f14a58ce57439314ff222d778b (openjdk:8u141-jdk-slim)
Tagging openjdk:8u141-jdk-slim
Tagging openjdk:8u141-slim
Tagging openjdk:8-jdk-slim
Tagging openjdk:8-slim
Tagging openjdk:jdk-slim
Tagging openjdk:slim

$ test/run.sh openjdk:8u141-jdk-slim
testing openjdk:8u141-jdk-slim
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'java-hello-world' [5/5]...passed


$ bashbrew build openjdk:8u131-jdk-alpine
Using bashbrew/cache:ed0268390b8d4051b764a6dbdac0d7d9df6a0c80384e860604e2f21c1c11d078 (openjdk:8u131-jdk-alpine)
Tagging openjdk:8u131-jdk-alpine
Tagging openjdk:8u131-alpine
Tagging openjdk:8-jdk-alpine
Tagging openjdk:8-alpine
Tagging openjdk:jdk-alpine
Tagging openjdk:alpine

$ test/run.sh openjdk:8u131-jdk-alpine
testing openjdk:8u131-jdk-alpine
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'java-hello-world' [5/5]...passed


$ bashbrew build openjdk:8u141-jdk-windowsservercore
skipping "openjdk:8u141-jdk-windowsservercore" (due to architecture "amd64"; only "windows-amd64" supported)

$ test/run.sh openjdk:8u141-jdk-windowsservercore
testing openjdk:8u141-jdk-windowsservercore
	image does not exist!


$ bashbrew build openjdk:8u141-jdk-nanoserver
skipping "openjdk:8u141-jdk-nanoserver" (due to architecture "amd64"; only "windows-amd64" supported)

$ test/run.sh openjdk:8u141-jdk-nanoserver
testing openjdk:8u141-jdk-nanoserver
	image does not exist!


$ bashbrew build openjdk:8u141-jre
Building bashbrew/cache:f2994a240934ee4cff120314f3bdd1c8bfbdede9e5309f0b93781a38e7080c30 (openjdk:8u141-jre)
Tagging openjdk:8u141-jre
Tagging openjdk:8-jre
Tagging openjdk:jre

$ test/run.sh openjdk:8u141-jre
testing openjdk:8u141-jre
	'utc' [1/6]...passed
	'cve-2014--shellshock' [2/6]...passed
	'no-hard-coded-passwords' [3/6]...passed
	'override-cmd' [4/6]...passed
	'java-hello-world' [5/6]...passed
	'java-uimanager-font' [6/6]...passed


$ bashbrew build openjdk:8u141-jre-slim
Using bashbrew/cache:e047b193b721160b2de0d4af6e2616e5e78bcd4ddc680a7385e104f78893b987 (openjdk:8u141-jre-slim)
Tagging openjdk:8u141-jre-slim
Tagging openjdk:8-jre-slim
Tagging openjdk:jre-slim

$ test/run.sh openjdk:8u141-jre-slim
testing openjdk:8u141-jre-slim
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'java-hello-world' [5/5]...passed


$ bashbrew build openjdk:8u131-jre-alpine
Using bashbrew/cache:1527af3b6e3ae489b01692c15f60ea0ac04f99651e5cdb217244addc522172cb (openjdk:8u131-jre-alpine)
Tagging openjdk:8u131-jre-alpine
Tagging openjdk:8-jre-alpine
Tagging openjdk:jre-alpine

$ test/run.sh openjdk:8u131-jre-alpine
testing openjdk:8u131-jre-alpine
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'java-hello-world' [5/5]...passed


$ bashbrew build openjdk:9-b181-jdk
Building bashbrew/cache:945c71fb39ce480b7048a4bef1c9707058e26fe16f7331e16186052143f1106c (openjdk:9-b181-jdk)
Tagging openjdk:9-b181-jdk
Tagging openjdk:9-b181
Tagging openjdk:9-jdk
Tagging openjdk:9

$ test/run.sh openjdk:9-b181-jdk
testing openjdk:9-b181-jdk
	'utc' [1/6]...passed
	'cve-2014--shellshock' [2/6]...passed
	'no-hard-coded-passwords' [3/6]...passed
	'override-cmd' [4/6]...passed
	'java-hello-world' [5/6]...passed
	'java-uimanager-font' [6/6]...passed


$ bashbrew build openjdk:9-b181-jdk-slim
Using bashbrew/cache:1e1dd3dc96752af20e4b4b8e2d1d42038c7e61b2ad38060677997c72071abe65 (openjdk:9-b181-jdk-slim)
Tagging openjdk:9-b181-jdk-slim
Tagging openjdk:9-b181-slim
Tagging openjdk:9-jdk-slim
Tagging openjdk:9-slim

$ test/run.sh openjdk:9-b181-jdk-slim
testing openjdk:9-b181-jdk-slim
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'java-hello-world' [5/5]...passed


$ bashbrew build openjdk:9-b154-jdk-windowsservercore
skipping "openjdk:9-b154-jdk-windowsservercore" (due to architecture "amd64"; only "windows-amd64" supported)

$ test/run.sh openjdk:9-b154-jdk-windowsservercore
testing openjdk:9-b154-jdk-windowsservercore
	image does not exist!


$ bashbrew build openjdk:9-b154-jdk-nanoserver
skipping "openjdk:9-b154-jdk-nanoserver" (due to architecture "amd64"; only "windows-amd64" supported)

$ test/run.sh openjdk:9-b154-jdk-nanoserver
testing openjdk:9-b154-jdk-nanoserver
	image does not exist!


$ bashbrew build openjdk:9-b181-jre
Building bashbrew/cache:6d7bef6fc942326c33becc1377b2c208d50ed2e3ec87d1bc7f9ff944480d045c (openjdk:9-b181-jre)
Tagging openjdk:9-b181-jre
Tagging openjdk:9-jre

$ test/run.sh openjdk:9-b181-jre
testing openjdk:9-b181-jre
	'utc' [1/6]...passed
	'cve-2014--shellshock' [2/6]...passed
	'no-hard-coded-passwords' [3/6]...passed
	'override-cmd' [4/6]...passed
	'java-hello-world' [5/6]...passed
	'java-uimanager-font' [6/6]...passed


$ bashbrew build openjdk:9-b181-jre-slim
Using bashbrew/cache:c0bb60e2a2bfad43fae768751e6115a6ed212e0f056892c10a835f47f78c599a (openjdk:9-b181-jre-slim)
Tagging openjdk:9-b181-jre-slim
Tagging openjdk:9-jre-slim

$ test/run.sh openjdk:9-b181-jre-slim
testing openjdk:9-b181-jre-slim
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'java-hello-world' [5/5]...passed

```


The following images failed at least one test:

- `openjdk`:
  - `openjdk:8u141-jdk-windowsservercore`
  - `openjdk:8u141-jdk-nanoserver`
  - `openjdk:9-b154-jdk-windowsservercore`
  - `openjdk:9-b154-jdk-nanoserver`

File: test/config.sh [hunk 1]
```diff
@@ -232,6 +233,10 @@ globalExcludeTests+=(
 	# https://github.com/docker-library/official-images/pull/1721#issuecomment-234128477
 	[clearlinux_no-hard-coded-passwords]=1
 
+	# alpine/slim openjdk images are non-headless and so can't do font stuff
```
Comment (tianon) [inline test/config.sh]: "alpine/slim openjdk images are non-headless" ?  you mean _are_ headless, right?

**Comment by tianon** [comment]
Yack yack yack; LGTM

**Comment by tianon** [comment]
image

**Pull request merged**